### PR TITLE
Fix RC overrides

### PR DESCRIPTION
**Pull request by peterbarker**
.... and add a simple test for the same.

I broke these with a bad rebase just before the move-channel-option-up stuff was merged.

### CODE REVIEW

File: Tools/autotest/apmrover2.py [hunk 1]
```diff
@@ -542,6 +542,66 @@ def test_setting_modes_via_auxswitches(self):
...
+        self.mavproxy.send("relay set 0 0\n")
+        off = self.get_parameter("SIM_PIN_MASK")
+        self.mavproxy.send("relay set 0 1\n")
+        on = self.get_parameter("SIM_PIN_MASK")
+        if on == off:
+            self.progress("Pin mask unchanged after relay command")
+            raise NotAchievedException()
+        self.progress("Pin mask changed after relay command")
+
+    def test_rc_overrides(self):
```
Comment (khancyr) [inline Tools/autotest/apmrover2.py]: should be in common, no ?

**Comment by peterbarker** [comment]
On Thu, 2 Aug 2018, WickedShell wrote:

> :( The indirection level... (silly subobjects). First question is can we actually solve this instead by leaving the old behaviour and call the super
> constructor to pass the channels pointer up?

We could store pointers in the base class, yes.  We don't do that for gcs 
or gcs_mavlink or arming or avoidance - if we're going to do it we should 
do it as a pattern.

> Shouldn't you be using rc_channel(chan) here? This seems to be an unneeded indirection level, and you have a helper designed for this.

I am planning on killing the statics at some stage - they're preserved so 
Plane didn't see massive amounts of churn.

In this case clear_override should be non-static - and possibly 
non-existant as the caller could probably make the 
rc().channel(3).clear_override() call itself (give or take sanity checks).

The rc_channel helper... well... I'm really not sure it adds that much.

**Comment by WickedShell** [comment]
> The rc_channel helper... well... I'm really not sure it adds that much.

I'm not sure it does either, in which case why have it at all? (I'm mostly just objecting to the multiple paths to the same output).

**Comment by peterbarker** [comment]
On Thu, 2 Aug 2018, WickedShell wrote:

>       The rc_channel helper... well... I'm really not sure it adds that much.
> 
> I'm not sure it does either, in which case why have it at all? (I'm mostly just objecting to the multiple paths to the same output).

Just the code-churn thing.  tridge might not have approved the PR if it 
changed 100+ lines in Plane...

**Pull request merged**